libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -620,11 +622,17 @@ internal class MapboxTripSession(
                     }
                 }
             }
-            checkVoiceInstructionEvent(progress.voiceInstructions) { voiceInstruction ->
-                voiceInstructionsObservers.forEach {
-                    it.onNewVoiceInstructions(voiceInstruction)
-                }
+            triggerVoiceInstructionEvent(progress)
+        }
+    }
+
+    private fun triggerVoiceInstructionEvent(progress: RouteProgress) {
+        val voiceInstructions = progress.voiceInstructions
+        if (voiceInstructions != null && lastVoiceInstruction != voiceInstructions) {
```

Comment:
I'm wondering if checking for equality here might actual lead to some edge case bugs.

For example, if there's a "turn right in 50 meters" instruction and after we take that turn the next one is also "turn right in 50 meters". In such a case, we wouldn't emit the second instruction and leave a driver without audio guidance.

I think that we shouldn't check for equality here and always trust NN that if they emit a non-null instruction then we should use it and emit it as well.

Reply:
this is a good point

Reply:
wait a minute. They should have different `distanceAlongGeometry`, shouldn't they? @LukasPaczos

Reply:
In an unfortunate case, that could be equal as well.

Reply:
Oh wait, maybe not :thinking:

Reply:
yep, you're right, they could be equals. _distanceAlongGeometry is the distance to next maneuver in unit meters_

Reply:
fixed